### ISSUE TITLE
Add a convenience functions for creating coverage plots 

### ISSUE THREAD

**Issue by tomkinsc** [broadinstitute/viral-ngs#402]
It would be helpful if viral-ngs could generate coverage plots. Two functions should be provided: one that accepts an aligned bam file and tabulates read depth (`plot_coverage()`?), and another (`align_and_plot_coverage()`?) that accepts an unaligned bam file or fastq files (converting where necessary), performs a quick alignment via`bwa mem`, and then calls the first function to calculate coverage.

The function `plot_coverage()` could use samtools' `sort`, `index`, and `depth` since they're fast, or it could use pysam. The former would include something like:

```
samtools sort -O bam -T sort mapped.bam > mapped_sorted.bam
samtools index mapped_sorted.bam
samtools depth -aa -m 1000000 mapped_sorted.bam > mapped_sorted_depth_summary.tsv
```

The function `plot_coverage()` should expose parameters to set the base quality and mapping thresholds (`-q` and `-Q`, respectively), and maximum coverage depth (`-m`). It should also expose parameters to set the plot type (png/pdf/txt), plot dimensions, title, and output path.

The function `align_and_plot_coverage()` should expose various bwa parameters such as min score threshold (`-T`). It could also expose a `--sensitive` option that uses @haydenm's param set, `-k 12 -A 1 -B 1 -O 1 -E 1 -T`, though that may mean [additional filtering is required](https://github.com/broadinstitute/hayden-work/blob/master/personal/bin/align_reads_to_fasta#L75).

Staying in Python space, if we're ok with including matplotlib as a dependency, we can use it to generate the coverage plots. Otherwise the most simple way to create plots would probably be `gnuplot`, which is available as a conda package. In addition to being able generating images or postscript plots, gnuplot supports a [fun text mode](http://www.calliopesounds.com/2011/12/i-have-to-say-again-i-find-gnuplots.html) suitable for terminal display, and an SVG mode suitable for display in a browser.

I'm not sure where `plot_coverage()` should live, but `read_utils.py` could make sense.

Once the functions have been added, the Snakemake rules should be updated to include them, possibly in `reports.rules`.

**Comment by haydenm** [comment]
Just a note that this gets tricky with multiple segments. Right now I sometimes concatenate segments (so just incorporating the third column of the `samtools depth` output), and sometimes separate segments into unique plots using `R`’s `facet_wrap`. I think ideally it would be a concatenation that lists the segment name below the x-axis; the information is there in the `samtools depth` output, where segment name is in the first column. It would just take some `matplotlib` sorcery.

**Comment by tomkinsc** [comment]
Great point. From a quick search, [this matplotlib plot](https://github.com/matted/genome_coverage_plotter) seems decent, though it may be nice to have axis breaks between segments and the labels below the plot.

**Comment by haydenm** [comment]
Yep! I definitely agree about putting the labels below the plot. Also, for a non-normalized plot (that one looks normalized), I think lines rather than dots look better, assuming the coverage is at least somewhat smooth.

**Comment by dpark01** [comment]
Concatenate the segments in order with no breaks in between: just color all the data points (or lines) separately by segment. That's how we plot everything in the eukaryotic world (ask Steve or Joe for examples). No legend or labels should be necessary for the segment/chromosome names because it should be inherently obvious, as long as each virus has a canonical ordering of their segments.

**Comment by tomkinsc** [comment]
Addressed in #403
